### PR TITLE
Use max changes

### PR DESCRIPTION
**Pull request by oubchid**
Related to #443

### CODE REVIEW

**Comment by oubchid** [comment]
Force-pushed for the rebase

File: src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js [hunk 1]
```diff
@@ -266,9 +266,10 @@ angular.module('esn.inbox.libs')
       return withJmapClient(function(jmapClient) {
         return jmapClient.mailbox_changes({
           accountId: null,
-          sinceState: inboxMailboxesCache.state
+          sinceState: inboxMailboxesCache.state,
+          maxChanges: MAILBOX_MAXIMUM_CHANGES_RESPONSE
         }).then(function(changes) {
-          if (changes.newState !== inboxMailboxesCache.state) {
+          if (changes.hasMoreChanges) {
```
Comment (alagane) [inline src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js:272]: This sems wrong. Even if `hasMoreChanges` is false, if the state has changed we should fetch mailboxes again.

File: src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.spec.js [hunk 2]
```diff
@@ -1563,7 +1564,7 @@ describe('The inboxMailboxesService factory', function() {
         });
       };
       jmapClient.mailbox_changes = function() {
-        return $q.when({ oldState: '1', newState: '2' });
+        return $q.when({ oldState: '1', newState: '2', hasMoreChanges: true });
```
Comment (alagane) [inline src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.spec.js:1567]: `hasMoreChanges` should not be true if `mailbox_get` return the same state than changes.newState `'2'`.

**Review by alagane — CHANGES_REQUESTED** [review]

File: src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js (hunk 1 above)
Comment (oubchid) [reply src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js:272]: so we should only compare the state diff

Comment (alagane) [reply src/esn.inbox.libs/app/services/mailboxes/mailboxes-service.js:272]: At least.
You can even take into account this issue: https://github.com/iNPUTmice/jmap/issues/47
If `hasMoreChanges` is true, anyway we can fetch all mailboxes (even if the state is unchanged)
And to go further, we can treat the error `cannotCalculateChanges` to fetch all mailboxes too.

**Comment by chibenwa** [comment]
@tuanlc would this qualify for a review?

**Review by tuanlc — COMMENTED** [review]
Hi,
- I cannot find the issue link in this PR so, hard for me to review when lacking the context
- Your commit titles are not good. A commit title should follow format: "#4xx use max changes", and if other commits just to fix the review, you need to mark them as "#fixup"

**Comment by oubchid** [comment]
> Hi,
> 
> * I cannot find the issue link in this PR so, hard for me to review when lacking the context
> * Your commit titles are not good. A commit title should follow format: "#4xx use max changes", and if other commits just to fix the review, you need to mark them as "#fixup"

Working actually on SmartSLA project will take note later

**Comment by chibenwa** [comment]
@alagane @fabienmoyon please do care about this piece of work.

**Pull request closed without merging**